demos/noise_suppression_demo/python/noise_suppression_demo.py

```diff
@@ -73,27 +74,36 @@ def main():
 
     log.info('OpenVINO Inference Engine')
     log.info('\tbuild: {}'.format(get_version()))
-    ie = IECore()
+    core = Core()
 
     # read IR
     log.info("Reading model {}".format(args.model))
-    model_xml = args.model
-    model_bin = model_xml.with_suffix(".bin")
-    ie_encoder = ie.read_network(model=model_xml, weights=model_bin)
+    ov_encoder = core.read_model(args.model)
 
     # check input and output names
-    input_shapes = {k: v.input_data.shape for k, v in ie_encoder.input_info.items()}
-    input_names = list(ie_encoder.input_info.keys())
-    output_names = list(ie_encoder.outputs.keys())
+    if len(ov_encoder.inputs) == len(ov_encoder.outputs):
+        input_shapes = {}
+        output_names = []
 
-    assert "input" in input_names, "'input' is not presented in model"
+        for const_obj in ov_encoder.inputs:
+            input_shapes[const_obj.get_names().pop()] = const_obj.shape
```

Comment:
anyway it will not help you to check input name, you need to take into account all possible names for tensor if you need specific name.

probably, better check in your case is:
```
try:
   data_input = model.input('input')
   input_shapes['input'] = data_input.shape
except RuntimeError:
   raise RuntimeError('input tensor with name input is not presented in the model')
```
and the same for main output

regarding states, you need to check get_names and select input/output where neccessary suffix exists (if not input/output and not state, then raise error)